Sources/SwiftProtobuf/Message+AsyncMessageSequence.swift

```diff
@@ -0,0 +1,102 @@
+//
+// Sources/SwiftProtobuf/Message+AsyncMessageSequence.swift - Async sequence over binary delimited protobuf
+//
+// Copyright (c) 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Extensions to `Message` to provide an async sequence over a binary delimited protobuf stream.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+@available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *)
+extension Message {
+  
+  /// Creates an asynchronous sequence of messages decoded from resource bytes
+  public static func asyncSequence(asyncBytes: URL.AsyncBytes) -> AsyncMessageSequence<Self> {
```

Comment:
To fully support users, you'll need the same optional arguments as on the delimited version: https://github.com/apple/swift-protobuf/blob/main/Sources/SwiftProtobuf/BinaryDelimited.swift#L117-L119

And then they will have to be applied to where you actually create the messages.

Reply:
Yes. This was a bit tricky because neither ExtensionMap nor BinaryDecodingOptions are Sendable. I implemented this using a factory to 1). avoid the double declaration of all the optional arguments and 2). so it could be marked @unchecked Sendable to avoid the warnings.

It seemed BinaryDecodingOptions can be made Sendable easily but then I ran into this:
https://github.com/apple/swift-protobuf/blob/main/Sources/SwiftProtobuf/MessageExtension.swift#L18

Reply:
It seems like adding `Sendable` to those generally would be a good thing since since just like you'd need it here, it seems like folks could need it in other places.

@Lukasa @FranzBusch @tbkka - thoughts?

Reply:
Sorry, I thought the issue was for `ExtensionMap` and `BinaryDecodingOptions`?

Reply:
Feel free to add Sendable to `ExtensionMap` `AnyMessageExtension` and `BinaryDecodingOptions` in this PR. These all are fine to require `Sendable` really.

Reply:
@thomasvl yes they're not ```Sendable``` but the warnings disappeared when I removed ```Sendable``` from  ```AsyncMessageSequence```

Reply:
The `AsyncMessageSequence` should really be `Sendable` though otherwise it looses a bunch of its functionality

Reply:
I haven't been able to make `AsyncMessageSequence` sendable without a compiler warning. Storing the `Base` sequence on this, as suggested earlier, gives a sendable warning.  I made all the optional param types `Sendable` anyway.

Reply:
Everything is now `Sendable` or `_ProtoSendable` as applicable.